app/mailers/caseworker_assignment_mailer.rb

```diff
@@ -0,0 +1,12 @@
+class CaseworkerAssignmentMailer < ApplicationMailer
```

Comment:
Hiya, not a proper review because I've only glanced through!

I think this class would be best named `CaseworkerMailer` based on the guidance here: https://katanacode.com/blog/ruby-on-rails-action-mailer-how-to-name-mailer-methods/

We could also follow their convention of naming the methods and update ours to be:
- `TeamLeaderMailer#project_created_notification`
- `CaseworkerMailer#caseworker_assigned_notification`

Let me know what you think!